### PR TITLE
added function: to_date_str

### PR DESCRIPTION
**Pull request by theodoreyzj**

### CODE REVIEW

File: src/test/scala/com/jakainas/functions/functionsTest.scala [hunk 1]
```diff
@@ -42,6 +42,23 @@ class functionsTest extends SparkTest {
     an [NullPointerException] should be thrownBy plusDays(null, 5)
   }
 
+  test("dateCols: numeric input"){
+    Seq(("2019", "04", "10")).toDF("year", "month", "day").select(dateCols('year, 'month, 'day))
+      .as[(String)].collect  should contain theSameElementsAs Array("2019-04-10")
```
Comment (fqaiser94) [inline src/test/scala/com/jakainas/functions/functionsTest.scala]: why the brackets `()` around `String`?

File: src/test/scala/com/jakainas/functions/functionsTest.scala [hunk 2]
```diff
@@ -42,6 +42,23 @@ class functionsTest extends SparkTest {
     an [NullPointerException] should be thrownBy plusDays(null, 5)
   }
 
+  test("dateCols: numeric input"){
```
Comment (fqaiser94) [inline src/test/scala/com/jakainas/functions/functionsTest.scala]: test titles should ideally be a bit more descriptive, something along the lines of what's being given and what should be returned e.g. 
`dateCols should return a properly formatted String date given valid numeric inputs`

File: src/test/scala/com/jakainas/functions/functionsTest.scala [hunk 3]
```diff
@@ -42,6 +42,23 @@ class functionsTest extends SparkTest {
...
+    Seq(("2019", "04", "10")).toDF("year", "month", "day").select(dateCols('year, 'month, 'day))
+      .as[(String)].collect  should contain theSameElementsAs Array("2019-04-10")
+
+    Seq((2019, 4, 10)).toDF("year", "month", "day").select(dateCols('year, 'month, 'day))
+      .as[String].collect  should contain theSameElementsAs Array("2019-04-10")
+  }
+
```
Comment (fqaiser94) [inline src/test/scala/com/jakainas/functions/functionsTest.scala]: what would happen if you gave dateCols invalid numeric input like `dateCols(2019, 13, 13)`?

File: src/main/scala/com/jakainas/functions/package.scala [hunk 4]
```diff
@@ -40,6 +40,18 @@ package object functions {
...
 
+  /**
+    * Generate a string Date column by combining year, month, and day columns.
+    * If one or more column is null, the result Date will be null.
+    * @param year - the year column
+    * @param month - the month column
+    * @param day - the day column
+    * @return
+    */
+  def dateCols(year: Column, month: Column, day: Column): Column = {
```
[the remark this answers is not in the document]
Comment (hearnden91) [reply src/main/scala/com/jakainas/functions/package.scala]: I don't get the 's'?  Its it returns a single StringType Column not multiple! ;) ...

Side: Why not include a `to_date` function in the lib that takes three Columns and returns a `DateType` in a Column?  

So for this function (assuming it returns StringType) -> 
1. As it does not return a "DateType", imo:  `date` _might_ be confusing?  
2. Perhaps:  `to_date_string`  or `genDateString`  or `mkDateString`  or `asDateString` ?

**Pull request merged**